client/components/language-picker/index.jsx

```diff
@@ -65,8 +67,9 @@ class LanguagePicker extends PureComponent {
 		} );
 	}
 
-	toggleOpen = () => {
+	onClick = ( event ) => {
```

Comment:
Thanks @spen for a great feedback! I extracted the toggling functionality back to a `toggleOpen` method and I'm calling it from the `handleClick` handler. I chose the `handleClick` name as it seems to be a [well established convention](http://redux-form.com/6.6.3/docs/faq/HandleVsOn.md/) in the React world :-)